File: Swift/프로그래머스/0/120866. 안전지대/안전지대.swift
import Foundation

func solution(_ board:[[Int]]) -> Int {
    let n = board.count
    var dangerZones = Array(repeating: Array(repeating: 0, count: n), count: n)
    let directions = [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1)]
    
    for i in 0..<n {
        for j in 0..<n {
            if board[i][j] == 1 {
                dangerZones[i][j] = 1
                for direction in directions {
                    let ni = i + direction.0
                    let nj = j + direction.1
                    if ni >= 0 && ni < n && nj >= 0 && nj < n {
                        dangerZones[ni][nj] = 1
                    }
                }
            }
        }
    }

    var safeCount = 0

    for i in 0..<n {
        for j in 0..<n {
            if dangerZones[i][j] == 0 && board[i][j] == 0 {
                safeCount += 1
            }
        }
    }

    return safeCount
}

File: Python3/프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    n = len(board)
    directions = [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 0), (0, 1), (1, -1), (1, 0), (1, 1) ]
    
    danger_zones = [[0] * n for _ in range(n)]
    
    for i in range(n):
        for j in range(n):
            if board[i][j] == 1:
                danger_zones[i][j] = 1
                
                for direction in directions:
                    ni, nj = i + direction[0], j + direction[1]
                    
                    if 0 <= ni < n and 0 <= nj < n:
                        danger_zones[ni][nj] = 1
                        
    safe_count = sum(danger_zones[i][j] == 0 for i in range(n) for j in range(n))
    
    return safe_count